### PR TITLE
Add vuid 06658, Validate image view sampler conversion format

### PR DESCRIPTION
**Pull request by ziga-lunarg**

### CODE REVIEW

**Review by sfricke-samsung — APPROVED** [review]

**Review by jeremyg-lunarg — COMMENTED** [review]
LGTM but failed internal CI. Was this change caught in an intermittent failure?

File: tests/vklayertests_buffer_image_memory_sampler.cpp [hunk 1]
```diff
@@ -15128,3 +15128,57 @@ TEST_F(VkLayerTest, TestCopyingInterleavedRegions) {
...
+    TEST_DESCRIPTION("Create image view with a different format that SamplerYcbcr was created with.");
+
+    SetTargetApiVersion(VK_API_VERSION_1_1);
+    ASSERT_NO_FATAL_FAILURE(InitFramework(m_errorMonitor));
+    if (DeviceValidationVersion() < VK_API_VERSION_1_1) {
+        printf("%s test requires Vulkan 1.1+, skipping test\n", kSkipPrefix);
+        return;
+    }
+
+    auto features11 = LvlInitStruct<VkPhysicalDeviceVulkan11Features>();
```
Comment (Tony-LunarG) [inline tests/vklayertests_buffer_image_memory_sampler.cpp:15142]: VkPhysicalDeviceVulkan11Features surprisingly needs VK_API_VERSION_1_2

File: tests/vklayertests_buffer_image_memory_sampler.cpp [hunk 2]
```diff
@@ -15128,3 +15128,57 @@ TEST_F(VkLayerTest, TestCopyingInterleavedRegions) {
...
+    sampler_conversion_ci.ycbcrRange = VK_SAMPLER_YCBCR_RANGE_ITU_FULL;
+    sampler_conversion_ci.components = {VK_COMPONENT_SWIZZLE_IDENTITY, VK_COMPONENT_SWIZZLE_IDENTITY, VK_COMPONENT_SWIZZLE_IDENTITY,
+                                        VK_COMPONENT_SWIZZLE_IDENTITY};
+    sampler_conversion_ci.xChromaOffset = VK_CHROMA_LOCATION_COSITED_EVEN;
+    sampler_conversion_ci.yChromaOffset = VK_CHROMA_LOCATION_COSITED_EVEN;
+    sampler_conversion_ci.chromaFilter = VK_FILTER_NEAREST;
+    sampler_conversion_ci.forceExplicitReconstruction = false;
+
+    VkSamplerYcbcrConversion sampler_conversion;
+    vk::CreateSamplerYcbcrConversion(device(), &sampler_conversion_ci, nullptr, &sampler_conversion);
```
Comment (Tony-LunarG) [inline tests/vklayertests_buffer_image_memory_sampler.cpp:15168]: The VkSamplerYcbcrConversion needs to be destroyed

**Review by Tony-LunarG — COMMENTED** [review]
LGTM with a couple of test nits

**Review by Tony-LunarG — APPROVED** [review]
LGTM

**Pull request merged**